From akpm@osdl.org Tue Mar 28 14:47:47 2006
Message-Id: <200603282247.k2SMlcVG029567@shell0.pdx.osdl.net>
From: Vivek Goyal <vgoyal@in.ibm.com>
Subject: PCI: 64 bit resources more drivers others changes
To: vgoyal@in.ibm.com, greg@kroah.com, mm-commits@vger.kernel.org
Date: Tue, 28 Mar 2006 14:47:32 -0800

From: Vivek Goyal <vgoyal@in.ibm.com>

- Changes required to fix compilation warnings for drivers/* dir for 64bit
  resources.  These changes came up due to cross-compilation on powerpc with
  CONFIG_PPC=32

Signed-off-by: Vivek Goyal <vgoyal@in.ibm.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/macintosh/macio_asic.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/macintosh/macio_asic.c
+++ gregkh-2.6/drivers/macintosh/macio_asic.c
@@ -428,10 +428,10 @@ static struct macio_dev * macio_add_one_
 
 	/* MacIO itself has a different reg, we use it's PCI base */
 	if (np == chip->of_node) {
-		sprintf(dev->ofdev.dev.bus_id, "%1d.%08lx:%.*s",
+		sprintf(dev->ofdev.dev.bus_id, "%1d.%016llx:%.*s",
 			chip->lbus.index,
 #ifdef CONFIG_PCI
-			pci_resource_start(chip->lbus.pdev, 0),
+			(unsigned long long)pci_resource_start(chip->lbus.pdev, 0),
 #else
 			0, /* NuBus may want to do something better here */
 #endif
